### PR TITLE
[FIX] account: prevent taxes on "payment_term" aml

### PR DESCRIPTION
**Pull request by kitan191**
Steps to reproduce:
- Go to Contacts and open any contact (e.g. Contact X)
- Check its configured Account Receivable (e.g. 121000 Account Receivable)
- Configure that account and set a default Sales tax on it (e.g. Tax 15%)
- Create an invoice with Contact X as customer
- Add an invoice line for any amount (e.g. $100) and any tax
- Swith to "Journal Items" tab and display "Taxes" column
A line (with 121000 Account Receivable) has been created automatically to balance the invoice line.
This line has a tax (i.e. Tax 15%): the default one configured on the account.
There should not be a tax for a receivable account.
Another issue is that the tax is set on the aml but not tax line has been created.
- Go back to "Invoice Lines" tab
- Remove the tax on the invoice line
- Save
In "Journal Items" tab, the tax line has been created from the aml with the receivable account.
- Go back to "Invoice Lines" tab
- Add a tax on the invoice line again
- Save
- Remove the tax on the invoice line
- Save
In "Journal Items" tab, there are now 2 tax lines from the aml with the receivable account.
By repeating the previous steps, new tax lines can be created infinitely.

Solution:
To prevent the issue, the default taxes shouldn't be populated on the aml when the account is a receivable (or payable) one and when the move type is an "invoice" type one.
As "display_type" field of "account.move.line" is set to "payment_term" when the account is a receivable or a payable one and when "is_invoice" is True on the move, no tax will be set in "_compute_tax_ids" computed method when "display_type" is "payment_term".

opw-3345953

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by dbkosky — APPROVED** [review]
This looks good to me. I've tested it locally and it works, and the test is a nice addition.
I'll forward this to @Levizar

**Comment by Levizar** [comment]
It seems legit but I will have to double check

File: addons/account/tests/test_account_move_out_invoice.py [hunk 1]
```diff
@@ -3692,3 +3692,20 @@ def test_post_valid_invoices_when_auto_post(self):
...
+        '''
+
+        receivable_account = self.partner_a.property_account_receivable_id
+        receivable_account.write({'tax_ids': [(6, 0, self.company_data['default_tax_sale'].ids)]})
+
+        move_form = Form(self.env['account.move'].with_context(default_move_type='out_invoice'))
+        move_form.partner_id = self.partner_a
+        with move_form.invoice_line_ids.new() as line_form:
+            line_form.product_id = self.product_a
+        move = move_form.save()
```
Comment (Levizar) [inline addons/account/tests/test_account_move_out_invoice.py:3707]: We should do a classic `self.env["account.move"].create(...)` as it doesn't really matter to use a `Form` here

File: addons/account/tests/test_account_move_out_invoice.py [hunk 2]
```diff
@@ -3692,3 +3692,20 @@ def test_post_valid_invoices_when_auto_post(self):
...
+
+        payment_term_line = move.line_ids.filtered(lambda l: l.display_type == 'payment_term')
+        self.assertEqual(payment_term_line.account_id.id, receivable_account.id)
+        self.assertEqual(len(payment_term_line.tax_ids), 0)
```
Comment (Levizar) [inline addons/account/tests/test_account_move_out_invoice.py:3711]: Not mandatory but we could use `assertRecordValues` to ass both at the same time

**Review by Levizar — CHANGES_REQUESTED** [review]
Good job!
Indeed that's something we should prevent.

**Comment by Levizar** [comment]
Also, I forgot to write it but we should rename the commit title to talk about payment_term line and not about receivable/payable

**Review by kitan191 — COMMENTED** [review]
Thanks for the review. :)

The suggested changes have been applied.

**Review by Levizar — APPROVED** [review]
Thanks :grin: 
@robodoo r+

**Pull request closed without merging**